app/exec/extension/_lib/vsix-manifest-builder.ts

```diff
@@ -396,6 +396,39 @@ export class VsixManifestBuilder extends ManifestBuilder {
 				}
 				this.addProperty("Microsoft.VisualStudio.Services.Content.Pricing.PriceCalculator", value.toString());
 				break;
+			case "galleryproperties":
+				/**
+				 * The Gallery Properties would be a generic array of JSON elements
+				 */
+				if (_.isArray(value)) {
+					value.forEach((propertyGroup: any) => {						
+						Object.keys(propertyGroup).forEach((propertyKey: string) => {
+							if(propertyKey && propertyGroup[propertyKey]) {
```

Comment:
Also, I'm not sure what this condition is actually trying to check. Can you make it more clear? For example, 

`if (typeof propertyKey === "string" && propertyKey.length > 0 && typeof propertyGroup[propertyKey] === "string" && propertyGroup[propertyKey].length > 0) { ... }`

Reply:
With the condition I just want to check that propertyKey and value are both non-null / not undefined and not empty.
I had on purpose not put a strict restriction on mandating the value to be a string. Also, since I am declaring propertyKey to be a string, I believe I don't need to check it?

Reply:
You are declaring it as a string, but there's no runtime type checking, so the result of parsing the JSON is not guaranteed to be consistent with your types. You need explicit runtime type checks.